Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/qrcode/QrCodeOptions.kt

```diff
@@ -0,0 +1,18 @@
+package de.rki.coronawarnapp.util.qrcode
+
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel
+import de.rki.coronawarnapp.util.HashExtensions.toSHA1
+import java.nio.charset.Charset
+
+/**
+ * @param correctionLevel [ErrorCorrectionLevel]
+ * @param characterSet [String]
+ */
+data class QrCodeOptions(
+    val correctionLevel: ErrorCorrectionLevel = ErrorCorrectionLevel.M,
+    val characterSet: Charset = Charsets.UTF_8,
+) {
+    val optionsKey by lazy {
+        "#$correctionLevel#$characterSet".toSHA1()
```

Comment:
maybe just the `hashCode` It is not really necessary to get `SHA1` right ?

Reply:
The `hashCode` is not guaranteed to be the same across different app sessions (jvm runs).
`Coil` expects a key that is consistent, e.g. to use during disk caching.